### DISCUSSION TITLE
Fastlane match: Is it possible to change the branch used to fetch the certificates?

### DISCUSSION THREAD

**Discussion by kiranrao-somnox** [fastlane/fastlane#21571] [Q&A]
Hello, I'm using fastlane match as follows:

- My app repository has `develop` and `release` branches.
- The certificates repo that is managed by fastlane match also has `develope` and `release` branches
- We use fastlane to upload to firebase from the `develop` branch and to upload to testflight from the `release` branch of our app repo.

Now I want to be able to do the following:

- I want to be able to upload to firebase from other branches as well.
- I want fastlane to use the certficates and profiles from `develop` branch of the certificates repo regardless of what branch we are in the app repo.
    - Except for `release` branch in which case I want to use the certificates from the `release` branch

Basically I'm looking for a way to ignore the current branch and instead use the `type` parameter of `fastlane match` to determine which branch of the certificates repo to fetch from.

Is this possible?

**Comment by mollyIV** [comment]
Hello @kiranrao-somnox 👋 

[`match`](https://docs.fastlane.tools/actions/match/) has a `git_branch` parameter that allows to specific git branch to use. For instance:

```ruby
match(
  git_url: git_url, # <-- URL to the git repo containing all the certificates
  git_branch: git_branch # <-- Specific git branch to use
)
```

Is that what you are looking for? 😉

**Reply by kiranrao-somnox** [reply]
Could you elaborate on what this `git_branch` is for? I assume it is referring to the branch in the _certificates_ repo, not the _app_ repo. Is that right?

IIRC I tried this and fastlane still mismatched the certificates and app type. This is a boiled down version of my Fastfile

```
match(
  git_branch: "development",
  app_identifier: "my.app.staging",
  type: "development",
)

match(
  git_branch: "release",
  app_identifier: "my.app.release",
  type: "appstore",
)
```

What is the expected behavior if I run `fastlane ios certificates` from, say, a `feature/login` branch oif my _app_ repo?
